### ISSUE TITLE
DB2 error while attempting to install Manage using one-click playbook

### ISSUE THREAD

**Issue by amkanjil** [ibm-mas/ansible-devops#552]
Same error with the latest version of mas_devops collection in ROKS and ROSA clusters. Please find the log below:

[root@AWN-IBMBDF-DR00 NBiTS_UNX_ENDUSR_2022_1]# /usr/bin/ansible-playbook ibm.mas_devops.oneclick_add_manage
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
[WARNING]: running playbook inside collection ibm.mas_devops
[WARNING]: ansible.utils.display.initialize_locale has not been called, this may result in incorrectly calculated text widths that can cause Display to print
incorrect line lengths

PLAY [localhost] *********************************************************************************************************************************************

TASK [Gathering Facts] ***************************************************************************************************************************************
ok: [localhost]

TASK [Check for required environment variables] **************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [ibm.mas_devops.db2 : Fail if required properties have not been provided] *******************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [ibm.mas_devops.db2 : include_tasks] ********************************************************************************************************************
included: /root/.ansible/collections/ansible_collections/ibm/mas_devops/roles/db2/tasks/determine-storage-classes.yml for localhost

TASK [ibm.mas_devops.db2 : Load default storage class information] *******************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Lookup storage classes] ***********************************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Debug available storage classes] **************************************************************************************************
ok: [localhost] => {
    "msg": [
        "efs",
        "gp2",
        "gp2-csi",
        "gp3",
        "gp3-csi"
    ]
}

TASK [ibm.mas_devops.db2 : Default Data Storage for ROKS if not set by user (ReadWriteOnce)] *****************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Assert that a data storage class has been defined] ********************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [ibm.mas_devops.db2 : Default Meta Storage for ROKS if not set by user (ReadWriteMany)] *****************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Assert that a meta storage class has been defined] ********************************************************************************
ok: [localhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [ibm.mas_devops.db2 : Default Logs Storage for ROKS if not set by user (ReadWriteOnce)] *****************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Default Backup Storage for ROKS if not set by user (ReadWriteMany)] ***************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Default Temp Storage for ROKS if not set by user (ReadWriteOnce)] *****************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Debug DB2U storage class configuration] *******************************************************************************************
ok: [localhost] => {
    "msg": [
        "Storage class (meta) ......... efs",
        "Storage class (data) ......... gp2",
        "Storage class (backup) ....... efs",
        "Storage class (logs) ......... gp2",
        "Storage class (temp) ......... gp2"
    ]
}

TASK [ibm.mas_devops.db2 : include_tasks] ********************************************************************************************************************
included: /root/.ansible/collections/ansible_collections/ibm/mas_devops/roles/db2/tasks/setup_norootsquash.yml for localhost

TASK [ibm.mas_devops.db2 : Set 'ibm-registry' secret content] ************************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Generate 'ibm-registry' secret] ***************************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Create 'norootsquash' DaemonSet] **************************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Wait for 'norootsquash' DaemonSet to be running on all nodes] *********************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Debug information] ****************************************************************************************************************
ok: [localhost] => {
    "msg": [
        "Namespace .................... db2u",
        "Storage class (meta) ......... efs",
        "Storage class (data) ......... gp2",
        "Storage class (backup) ....... efs",
        "Storage class (logs) ......... gp2",
        "Storage class (temp) ......... gp2",
        "Db2 database instanceId ...... masdb1",
        "Db2 database name ............ BLUDB",
        "Db2 version .................. 11.5.7.0-cn4",
        "MAS Instance ID .............. rosa-mas89",
        "MAS Config directory ......... /root/MAS89/config"
    ]
}

TASK [ibm.mas_devops.db2 : Create db2u Namespace] ************************************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Set 'ibm-registry' secret content] ************************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Generate 'ibm-registry' secret] ***************************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Create db2u Operand Request] ******************************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Get cluster subdomain] ************************************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Create internal CA certificate issuer] ********************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Create and wait for CA certificate] ***********************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Create certificate issuer] ********************************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Create db2u certificate] **********************************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Taint and label dedicated worker node] ********************************************************************************************
skipping: [localhost]

TASK [ibm.mas_devops.db2 : Wait until the Db2uCluster CRD is available] **************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : See if db2u instance already exists] **********************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : Create db2 instance] **************************************************************************************************************
changed: [localhost]

TASK [ibm.mas_devops.db2 : Wait for db2u instance to be ready (5m delay)] ************************************************************************************
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (24 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (23 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (22 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (21 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (20 retries left).

FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (19 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (18 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (17 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (16 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (15 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (14 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (13 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (12 retries left).

FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (11 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (10 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (9 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (8 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (7 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (6 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (5 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (4 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (3 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (2 retries left).
FAILED - RETRYING: Wait for db2u instance to be ready (5m delay) (1 retries left).
fatal: [localhost]: FAILED! => {"api_found": true, "attempts": 24, "changed": false, "resources": [{"apiVersion": "db2u.databases.ibm.com/v1", "kind": "Db2uCluster", "metadata": {"annotations": {"db2u/certs-api-cert": "[secure]", "db2u/certs-api-key": "[secure]", "db2u/certs-wv-rest": "[secure]", "db2u/license": "[secure]", "db2u/sshkeys-db2instusr": "[secure]", "db2u/sshkeys-db2uadm": "[secure]", "db2u/sshkeys-db2uhausr": "[secure]", "kubectl.kubernetes.io/last-applied-configuration": "{\"apiVersion\":\"db2u.databases.ibm.com/v1\",\"kind\":\"Db2uCluster\",\"metadata\":{\"name\":\"masdb1\",\"namespace\":\"db2u\"},\"spec\":{\"account\":{\"imagePullSecrets\":[\"ibm-registry\"],\"privileged\":true},\"addOns\":{\"graph\":{\"enabled\":false},\"rest\":{\"enabled\":false}},\"environment\":{\"database\":{\"dbConfig\":{\"APPLHEAPSZ\":\"8192 AUTOMATIC\"},\"name\":\"BLUDB\",\"settings\":{\"dftTableOrg\":\"ROW\"},\"ssl\":{\"certLabel\":\"CN=db2u\",\"secretName\":\"db2u-certificate\"}},\"dbType\":\"db2wh\",\"instance\":{\"dbmConfig\":{\"INSTANCE_MEMORY\":\"AUTOMATIC\"},\"registry\":{\"DB2AUTH\":\"OSAUTHDB,ALLOW_LOCAL_FALLBACK,PLUGIN_AUTO_RELOAD\",\"DB2_4K_DEVICE_SUPPORT\":\"ON\",\"DB2_FMP_RUN_AS_CONNECTED_USER\":\"NO\",\"DB2_WORKLOAD\":\"ANALYTICS\"}},\"mln\":{\"total\":1}},\"license\":{\"accept\":true},\"podConfig\":{\"db2u\":{\"resource\":{\"db2u\":{\"limits\":{\"cpu\":\"6000m\",\"memory\":\"12Gi\"},\"requests\":{\"cpu\":\"4000m\",\"memory\":\"8Gi\"}}}}},\"size\":1,\"storage\":[{\"name\":\"meta\",\"spec\":{\"accessModes\":[\"ReadWriteMany\"],\"resources\":{\"requests\":{\"storage\":\"20Gi\"}},\"storageClassName\":\"efs\"},\"type\":\"create\"},{\"name\":\"data\",\"spec\":{\"accessModes\":[\"ReadWriteOnce\"],\"resources\":{\"requests\":{\"storage\":\"100Gi\"}},\"storageClassName\":\"gp2\"},\"type\":\"template\"},{\"name\":\"backup\",\"spec\":{\"accessModes\":[\"ReadWriteMany\"],\"resources\":{\"requests\":{\"storage\":\"100Gi\"}},\"storageClassName\":\"efs\"},\"type\":\"create\"},{\"name\":\"activelogs\",\"spec\":{\"accessModes\":[\"ReadWriteOnce\"],\"resources\":{\"requests\":{\"storage\":\"100Gi\"}},\"storageClassName\":\"gp2\"},\"type\":\"template\"},{\"name\":\"tempts\",\"spec\":{\"accessModes\":[\"ReadWriteOnce\"],\"resources\":{\"requests\":{\"storage\":\"100Gi\"}},\"storageClassName\":\"gp2\"},\"type\":\"template\"}],\"version\":\"11.5.7.0-cn4\"}}"}, "creationTimestamp": "2022-12-07T06:51:19Z", "generation": 4, "managedFields": [{"apiVersion": "db2u.databases.ibm.com/v1", "fieldsType": "FieldsV1", "fieldsV1": {"f:metadata": {"f:annotations": {"f:db2u/certs-api-cert": {}, "f:db2u/certs-api-key": {}, "f:db2u/certs-wv-rest": {}, "f:db2u/license": {}, "f:db2u/sshkeys-db2instusr": {}, "f:db2u/sshkeys-db2uadm": {}, "f:db2u/sshkeys-db2uhausr": {}}}, "f:spec": {"f:addOns": {"f:qrep": {".": {}, "f:infraHost": {}, "f:infraIP": {}, "f:license": {}}}, "f:environment": {"f:instance": {"f:password": {}}, "f:ldap": {".": {}, "f:blueAdminPassword": {}, "f:enabled": {}, "f:password": {}}}}, "f:status": {".": {}, "f:maintenanceState": {}, "f:state": {}}}, "manager": "manager", "operation": "Update", "time": "2022-12-07T06:52:38Z"}, {"apiVersion": "db2u.databases.ibm.com/v1", "fieldsType": "FieldsV1", "fieldsV1": {"f:status": {"f:conditions": {}, "f:version": {}}}, "manager": "manager", "operation": "Update", "subresource": "status", "time": "2022-12-07T06:52:38Z"}, {"apiVersion": "db2u.databases.ibm.com/v1", "fieldsType": "FieldsV1", "fieldsV1": {"f:metadata": {"f:annotations": {".": {}, "f:kubectl.kubernetes.io/last-applied-configuration": {}}}, "f:spec": {".": {}, "f:account": {".": {}, "f:imagePullSecrets": {}, "f:privileged": {}}, "f:addOns": {".": {}, "f:graph": {".": {}, "f:enabled": {}}, "f:rest": {".": {}, "f:enabled": {}}}, "f:environment": {".": {}, "f:database": {".": {}, "f:dbConfig": {".": {}, "f:APPLHEAPSZ": {}}, "f:name": {}, "f:settings": {".": {}, "f:dftTableOrg": {}}, "f:ssl": {".": {}, "f:certLabel": {}, "f:secretName": {}}}, "f:dbType": {}, "f:instance": {".": {}, "f:dbmConfig": {".": {}, "f:INSTANCE_MEMORY": {}}, "f:registry": {".": {}, "f:DB2AUTH": {}, "f:DB2_4K_DEVICE_SUPPORT": {}, "f:DB2_FMP_RUN_AS_CONNECTED_USER": {}, "f:DB2_WORKLOAD": {}}}, "f:mln": {".": {}, "f:total": {}}}, "f:license": {".": {}, "f:accept": {}}, "f:podConfig": {".": {}, "f:db2u": {".": {}, "f:resource": {".": {}, "f:db2u": {".": {}, "f:limits": {".": {}, "f:cpu": {}, "f:memory": {}}, "f:requests": {".": {}, "f:cpu": {}, "f:memory": {}}}}}}, "f:size": {}, "f:storage": {}, "f:version": {}}}, "manager": "OpenAPI-Generator", "operation": "Update", "time": "2022-12-07T09:02:09Z"}], "name": "masdb1", "namespace": "db2u", "resourceVersion": "826327", "uid": "5b7dd0ff-1e29-4e93-8b44-625b3b289978"}, "spec": {"account": {"imagePullSecrets": ["ibm-registry"], "privileged": true}, "addOns": {"graph": {"enabled": false}, "qrep": {"infraHost": "xxx.ibm.com", "infraIP": "x.x.x.x", "license": {}}, "rest": {"enabled": false}}, "environment": {"database": {"dbConfig": {"APPLHEAPSZ": "8192 AUTOMATIC"}, "name": "BLUDB", "settings": {"dftTableOrg": "ROW"}, "ssl": {"certLabel": "CN=db2u", "secretName": "db2u-certificate"}}, "dbType": "db2wh", "instance": {"dbmConfig": {"INSTANCE_MEMORY": "AUTOMATIC"}, "password": "[secure]", "registry": {"DB2AUTH": "OSAUTHDB,ALLOW_LOCAL_FALLBACK,PLUGIN_AUTO_RELOAD", "DB2_4K_DEVICE_SUPPORT": "ON", "DB2_FMP_RUN_AS_CONNECTED_USER": "NO", "DB2_WORKLOAD": "ANALYTICS"}}, "ldap": {"blueAdminPassword": "[secure]", "enabled": true, "password": "[secure]"}, "mln": {"total": 1}}, "license": {"accept": true}, "podConfig": {"db2u": {"resource": {"db2u": {"limits": {"cpu": "6000m", "memory": "12Gi"}, "requests": {"cpu": "4000m", "memory": "8Gi"}}}}}, "size": 1, "storage": [{"name": "meta", "spec": {"accessModes": ["ReadWriteMany"], "resources": {"requests": {"storage": "20Gi"}}, "storageClassName": "efs"}, "type": "create"}, {"name": "data", "spec": {"accessModes": ["ReadWriteOnce"], "resources": {"requests": {"storage": "100Gi"}}, "storageClassName": "gp2"}, "type": "template"}, {"name": "backup", "spec": {"accessModes": ["ReadWriteMany"], "resources": {"requests": {"storage": "100Gi"}}, "storageClassName": "efs"}, "type": "create"}, {"name": "activelogs", "spec": {"accessModes": ["ReadWriteOnce"], "resources": {"requests": {"storage": "100Gi"}}, "storageClassName": "gp2"}, "type": "template"}, {"name": "tempts", "spec": {"accessModes": ["ReadWriteOnce"], "resources": {"requests": {"storage": "100Gi"}}, "storageClassName": "gp2"}, "type": "template"}], "version": "11.5.7.0-cn4"}, "status": {"conditions": [{"lastTransitionTime": "2022-12-07T06:52:38Z", "status": "Processing", "type": "FormationStatus"}], "maintenanceState": "None", "state": "NotReady", "version": "11.5.7.0-cn4"}}]}

**Comment by whitfiea** [comment]
@amkanjil Did the db2ucluster instance finally become ready or did it just not get configured? Have you contacted DB2 support?

**Comment by aroute** [comment]
Likely a duplicate of [503](https://github.com/ibm-mas/ansible-devops/issues/503). It seems to happen as task moves to db2 config while the db2 instance is getting bounced (only happens when Manage is being deployed first before the IoT/Monitor). Not sure how this issue is being countered via the CICD pipeline.

**Comment by durera** [comment]
This is just reporting that the Db2 operator was unable to get the operand ready in the timeout that the automation provides.

```
"status": {"conditions": [{"lastTransitionTime": "2022-12-07T06:52:38Z", "status": "Processing", "type": "FormationStatus"}], "maintenanceState": "None", "state": "NotReady", "version": "11.5.7.0-cn4"}}]}
```

Per @whitfiea 's earlier comment, on seeing something like this I would either expect to see that there are some errors being spit out of the db2u operator (`db2u-operator-manager-xxxxx-xxxx` pod in the `ibm-common-services` namespace), or it's a case that the operator and/or the cluster is on a go slow.

Once the Kubernetes API accepts the Db2uCluster spec that we have generated and we start the `Wait for db2u instance to be ready` loop all the work is happening in the Db2u operator.

**Comment by durera** [comment]
The fix for #604 will likely go a long way to addressing issues with Db2 install reliability like this, as Db2 will now correctly know how much memory it has available to it.
